optimum/utils/input_generators.py

```diff
@@ -628,3 +629,46 @@ def generate(self, input_name: str, framework: str = "pt"):
             shape = [self.batch_size]
 
         return self.random_int_tensor(shape, max_value=max_value, framework=framework)
+
+
+class DummyVisualBertInputGenerator(DummyTextInputGenerator):
+    SUPPORTED_INPUT_NAMES = ("visual_embeds", "visual_token_type_ids", "visual_attention_mask","input_ids", "attention_mask","token_type_ids")
+    # todo: see how to add ,"region_to_phrase_position" since that input name raises error (ValueError: Config dummy inputs are not a subset of the model inputs)
+    def __init__(
+        self,
+        task: str,
+        normalized_config: NormalizedTextConfig,
+        visual_seq_length: int = DEFAULT_DUMMY_SHAPES["visual_seq_length"],
+        batch_size: int = DEFAULT_DUMMY_SHAPES["batch_size"],
+        sequence_length: int = DEFAULT_DUMMY_SHAPES["sequence_length"],
+        num_choices: int = DEFAULT_DUMMY_SHAPES["num_choices"],
+    ):
+
+        super().__init__(task, normalized_config)
+        self.visual_seq_length = visual_seq_length
+
+    def generate(self, input_name: str, framework: str = "pt"):
+        # TODO maybe the following should be checked with the model checkpoint_path and checking existance of for example "vqa" substring is better
+        if self.task in ["default", "visual-question-answering", "region-to-phrase-alignment",]:
+            visual_embedding_dim = 2048
+        elif self.task == "multiple-choice":
+            visual_embedding_dim = 512
+        elif self.task == "natural-language-for-visual-reasoning":
+            visual_embedding_dim = 1024
```

Comment:
I agree with your comment, I think the input `visual_embedding_dim` should rather be passed in the `__init__` and depend on the model checkpoint.

I think a way to do this is to redefine in `VisualBertOnnxConfig` an attribute like `dynamic_input_shape = {"visual_embedding_dim": self._config.visual_embedding_dim}` in the init (and call the super() init).

Then, in `generate_dummy_inputs`, you could add something like:

```python
if self.hasattr("dynamic_input_shape"):
    for key in self.dynamic_input_shape:
        kwargs[key] = getattr(self, key)
```